### PR TITLE
[master] [KOGITO-1627] Load default test config

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-1627\

This will allow PR and nightlies to load against predefined version (case of cr builds).
This can still be overloaded when calling the script if needed.

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by Sgitario** [comment]
If I understood correctly, we can now have a file where to set the default config (".default_test_parameters"). This configuration was being set via parameters so far. 

With the parameter --ignore_default_config, we can ignore this file and load the parameters as we were doing so far. 

I wanted to add a suggestion here. Do we need this new parameter "ignore_default_config"? In order to avoid to add another parameter, what about this behaviour:
- We load **always** the .default_test_parameters file.
- We overwrite the parameters within the file by the ones set via the parameters

For example, if the default test parameters file contains:

```properties
tests.build-image-version=0.9.0-rc2
tests.services-image-version=0.9.0-rc2
tests.examples-uri=https://github.com/kiegroup/kogito-examples.git
tests.examples-ref=master
```

And, as an example, run the tests with the parameter --build-image-version=0.10.0

The final run will be executed using the parameters:

```properties
tests.build-image-version=0.10.0
tests.services-image-version=0.9.0-rc2
tests.examples-uri=https://github.com/kiegroup/kogito-examples.git
tests.examples-ref=master
```

I think it could be more intuitive this way and get rid of an extra parameter. However, this is only one suggestion!

File: test/.default_test_parameters [hunk 1]
```diff
@@ -0,0 +1,4 @@
+tests.build-image-version=0.9.0-rc2
+tests.services-image-version=0.9.0-rc2
```
Comment (sutaakar) [inline test/.default_test_parameters:2]: Is there a specific reason for using rc2 as 0.9.0 is released already?

Comment (radtriste) [reply test/.default_test_parameters:2]: jobs-service 0.9.0 does not work...

Comment (radtriste) [reply test/.default_test_parameters:2]: I could set directly the jobs-service 0.9.0-rc2 and let the others to 0.9.0

Comment (sutaakar) [reply test/.default_test_parameters:2]: could be good, also please add a comment explaining this setup and when it can be removed

**Comment by radtriste** [comment]
@MarianMacik I could set the default into Makefile indeed but I am not sure it should be in there to IMO (we could also remove the other parameters already set or migrate to config file)
@Sgitario I see your point but what if we want to test the default case, means no extra configuration (do not set the images version and examples repository is on branch `stable`)

for me that file is mainly for the pipeline because:
- the branches (master & 0.9.x) are already 0.10 or 0.9.1 and images to that version are not existing.
- also for the examples repository to branch 0.9.x and master for the tests

**Comment by radtriste** [comment]
Or I set just a pipeline config and have a `--load_default_config` parameter instead that would be used by the pipeline.

**Comment by Sgitario** [comment]
The problem I see is how the parameters and the values in the file coexist altogether. For example, if I don't know about the ignore parameter and I'm trying to overwrite the versions tag, it won't work. Therefore, I think having a "--load_default_config" as you suggested is more intuitive than having the "--ignore_default_config". But again, this is my point of view.

**Comment by radtriste** [comment]
>  For example, if I don't know about the ignore parameter and I'm trying to overwrite the versions tag, it won't work.

It will. Default config parameters are set only if not existing already
https://github.com/radtriste/kogito-cloud-operator/blob/12bfd0a69f4236d2a8002e7de31d54b337566b0d/hack/run-tests.sh#L294

but yeah the load_default_config is maybe better.

File: test/.default_test_parameters (hunk 1 above)
Comment (radtriste) [reply test/.default_test_parameters]: updated

**Comment by radtriste** [comment]
@sutaakar @MarianMacik @Sgitario can you have another look please ?
Changed to `load_default_config`

Makefile => Contains a default configuration for tests parameters, but no version information
.default_test_config => Contains version information. Need to use `load_default_config=true`. Can still be overridden via other parameters from the `make` command

File: deploy/examples/jobs-service.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+apiVersion: app.kiegroup.org/v1alpha1
+kind: KogitoJobsService
```
Comment (MarianMacik) [inline deploy/examples/jobs-service.yaml:2]: Is this related to the issue?

Comment (radtriste) [reply deploy/examples/jobs-service.yaml:2]: nope but it was missing. Can put it in another PR is needed.

**Comment by ricardozanini** [comment]
@radtriste could you please take a look at `hack/bump-version.sh` to make sure that your newly added file with the versions is covered?

**Comment by radtriste** [comment]
quite tricky as we could have very specific configuration but will have a look

File: deploy/examples/jobs-service.yaml (hunk 2 above)
Comment (MarianMacik) [reply deploy/examples/jobs-service.yaml:2]: Sure, np.

**Comment by radtriste** [comment]
@ricardozanini updated `bump-versions.sh`.
for a new version, it will override the configuration with brand new values :)
if anything to keep, we will be able to see into the PR anyway but I doubt it when there is a new PR.
one thing will be that the pipeline will be green only if images are already existing in the defined version and the branch for kogito-examples exist

File: deploy/examples/jobs-service.yaml (hunk 2 above)
Comment (ricardozanini) [reply deploy/examples/jobs-service.yaml:2]: Actually we don't need this since we have: https://github.com/kiegroup/kogito-cloud-operator/blob/master/deploy/crds/app.kiegroup.org_v1alpha1_kogitojobsservice_cr.yaml

But since we have examples for all services, I don't see any problems having it.

**Comment by ricardozanini** [comment]
Ready?

**Comment by radtriste** [comment]
not yet

**Comment by radtriste** [comment]
Final test passed
